sycl/source/detail/context_impl.cpp

```diff
@@ -104,6 +104,8 @@ cl_uint context_impl::get_info<info::context::reference_count>() const {
       this->getHandleRef());
 }
 template <> platform context_impl::get_info<info::context::platform>() const {
+  if (is_host())
+    return platform();
   return createSyclObjFromImpl<platform>(MPlatform);
```

Comment:
Suggest fixing it it a different way: properly initialize objects(like platform here) for the host device to avoid this workaround.

Reply:
I'm OK if you do it as a separate patch.

Reply:
will do rework in separate patch